azurecompute-arm/src/test/java/org/jclouds/azurecompute/arm/features/VirtualMachineApiLiveTest.java

```diff
@@ -0,0 +1,269 @@
+   public void testStart() {
+      api().start(getName());
+      String status = "";
+      int stopCounter = 0;
+      while (!status.equals("VM running")) {
+         List<VirtualMachineInstance.VirtualMachineStatus> statuses = api().getInstanceDetails(getName()).statuses();
+         for (int c = 0; c < statuses.size(); c++) {
+            if (statuses.get(c).code().substring(0, 10).equals("PowerState")) {
+               status = statuses.get(c).displayStatus();
+            }
+            try {
+               Thread.sleep(5 * 1000); // Starting takes long time. Recheck later
+            } catch (InterruptedException e) {
+               e.printStackTrace();
+            }
+            stopCounter++;
+            if (stopCounter == 40)
+               break;
+         }
+      }
+      assertEquals(status, "VM running");
+
+   }
+
+   @Test(dependsOnMethods = "testStop")
+   public void testRestart() {
+      api().start(getName());
+      String status = "";
+      int stopCounter = 0;
+      while (!status.equals("VM running")) {
+         List<VirtualMachineInstance.VirtualMachineStatus> statuses = api().getInstanceDetails(getName()).statuses();
+         for (int c = 0; c < statuses.size(); c++) {
+            if (statuses.get(c).code().substring(0, 10).equals("PowerState")) {
+               status = statuses.get(c).displayStatus();
+            }
+            try {
+               Thread.sleep(5 * 1000); // Starting takes long time. Recheck later
+            } catch (InterruptedException e) {
+               e.printStackTrace();
+            }
+            stopCounter++;
+            if (stopCounter == 40)
+               break;
+         }
+      }
+      api().restart(getName());
+      try {
+         Thread.sleep(30 * 1000); // Wait while to get it powered off and then wait until machine is running
+      } catch (InterruptedException e) {
+         e.printStackTrace();
+      }
+      status = "";
+      stopCounter = 0;
+      while (!status.equals("VM running")) {
+         List<VirtualMachineInstance.VirtualMachineStatus> statuses = api().getInstanceDetails(getName()).statuses();
+         for (int c = 0; c < statuses.size(); c++) {
+            if (statuses.get(c).code().substring(0, 10).equals("PowerState")) {
+               status = statuses.get(c).displayStatus();
+            }
+            try {
+               Thread.sleep(5 * 1000); // Starting takes long time. Recheck later
+            } catch (InterruptedException e) {
+               e.printStackTrace();
+            }
+            stopCounter++;
+            if (stopCounter == 40)
+               break;
+         }
+      }
+      assertEquals(status, "VM running");
+   }
+
+   @Test(dependsOnMethods = "testCreate")
+   public void testList() {
+      List<VirtualMachine> list = api().list();
+      for (VirtualMachine machine : list) {
+         assertTrue(!machine.name().isEmpty());
+      }
+   }
+
+   @Test(dependsOnMethods = "testRestart")
```

Comment:
Also `alwaysRun = true`? This will need to depend on other methods too, owhterwise this could be run *before* the list or get tests.